app/assets/javascripts/embeds.js

```diff
@@ -0,0 +1,81 @@
+/*
+  This script is used for receiving messages from iframes and updating
+  settings in the ETM. It uses the postMessage browser API for receiving cross
+  domain messages.
+  To add an additional message handler all you have to do is add a function to
+  the actions object. Be aware that an action can only be provided with a single
+  argument. The however can be an array or map but can't (or shouldn't) contain
+  functions.
+
+  The messages have to conform to a specific schema.
+    - An action key is required and should match an action declared in this file.
+    - An argument key with value is required and its schema is dependent on the
+      matching action below.
+
+  Below is an example message.
+
+  * {
+  *   action: "updateInlandWindTurbine",
+  *   argument: {
+  *     power: pico.inMegaWatt(power)
+  *   }
+  * }
+
+*/
+
+embeds = {
+  attach: function(){
+    window.addEventListener('message', embeds.receiveMessage, false)
+  },
+
+  // dependencies:
+  //   - jquery
+  //   - fancybox
+  close: function(event){
+    $.fancybox.close()
```

Comment:
I think this needs to also remove the "message" event that was added in `attach`.

```suggestion
    $.fancybox.close()
    window.removeEventListener('message', embeds.receiveMessage)
```

Reply:
> Look mostly good to me, Hound's nagging aside. 😆
> 
> Two minor comments:
> 
>     * Clicking the PICO link in an unsupported region (e.g. France) brings up PICO for the Netherlands. Could we instead check that the area is supported, and if not display a message informing the user?
good idea. I'll add it after I've taken care of 🐶 

>     * When ready to merge, would you please squash this down to remove the "work in progress" commits? e.g. • Update Webmock for better Ruby 2.6 support  • Added first version of PICO integration  • Any other commits which are relevant as separate commits

I was planning on squashing it all down to a single commit later on. These commits are for review purposes!